packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -123,70 +140,84 @@ export const DatePicker: React.VFC<DatePickerProps> = ({
+    /**
+     * react-day-picker will fire events for disabled days by default.
+     * We're checking here if it includes the CSS Modules class for disabled
+     * on the modifier to then return early.
+     */
     if (Object.keys(modifiers).includes(defaultCalendarClasses.disabled)) {
       return
     }
-    onChange(day)
+    onDayChange(day)
     setIsOpen(false)
   }
 
-  const handleOpenClose = () => {
-    setIsOpen(!isOpen)
+  const handleInputChange = (input: Date | undefined): void => {
+    if (input === undefined) {
+      onDayChange(undefined)
+      return
+    }
+
+    if (isInvalidDate(input) || isDisabledDate(input, disabledDays)) {
+      onDayChange(undefined)
+      return
+    }
+    onDayChange(input)
   }
 
-  const handleReturnFocus = () => {
+  const handleOpenClose = (): void => setIsOpen(!isOpen)
+
+  const handleReturnFocus = (): void => {
     if (buttonRef.current) {
       buttonRef.current.focus()
     }
   }
 
-  const disabledDays = calculateDisabledDays(
-    disabledDates,
-    disabledDaysOfWeek,
-    disabledRange,
-    disabledBeforeAfter,
-    disabledBefore,
-    disabledAfter
-  )
+  const handleButtonClick = (): void => {
+    handleOpenClose()
+    onButtonClick && onButtonClick()
+  }
+
+  const handleKeyDown = (
+    e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<Element>
+  ): void => {
+    // check whether event key is arrow down or alt + arrow down, open calendar if so.
+    if (e.key === "ArrowDown" || (e.key === "ArrowDown" && e.altKey === true)) {
+      e.preventDefault()
+      setIsOpen(true)
+    }
+  }
 
   return (
     <div ref={wrapperRef}>
       <div ref={setReferenceElement}>
-        <Label disabled={isDisabled} htmlFor={id} labelText={labelText} />
-        <button
-          className={cx(
-            datePickerStyles.button,
-            datePickerStyles.withStartIconAdornment,
-            {
-              [datePickerStyles.disabled]: isDisabled,
-            }
-          )}
+        <DateInput
           id={id}
+          inputRef={inputRef}
+          buttonRef={buttonRef}
+          isOpen={isOpen}
           disabled={isDisabled}
-          ref={buttonRef}
-          onClick={handleOpenClose}
-          aria-label={
-            value
-              ? `Change date: ${value.toLocaleDateString(
-                  "en-US",
-                  dateFormatOptions
-                )}`
-              : "Choose date"
-          }
-          {...inputProps}
-        >
-          <div className={datePickerStyles.startIconAdornment}>
-            <Icon icon={dateStart} role="presentation" />
-          </div>
-          <span className={datePickerStyles.value}>
-            {value ? value.toLocaleDateString("en-US", dateFormatOptions) : ""}
-          </span>
-        </button>
+          onBlur={handleInputChange}
+          labelText={labelText}
+          description="Format: mm/dd/yyyy"
```

Comment:
I had requested that this was left out of the consumers hands for a11y reasons. It's vital that we have the format there for users to know what format to enter a date in.

Reply:
Hmm. However, I can see someone wanting to write a description for the field.
Could we put the format into the aria-description somehow perhaps (when description is different, return description + format)? Although of course that opens us up for the possibility of doubling the format.

Reply:
If we wanted the ability to override, I'd recommend we name the prop something like `descriptionAndIUnderstandTheAccessibilityImplications` (which exists in another place for `newTabAndIUnderstandTheAccessibilityImplications`), though it's not just an a11y concern but a usability concern in general.

Reply:
Or we just always have this string attached.
```
description={`${description} (Format: mm/dd/yy)`}
```
or similar

Reply:
Yeah that could work also.

I wouldn't go for the aria-describedby solution because it's not just a concern for screen readers (sorry my bad for initially saying that is an a11y concern)

Reply:
The `nAndIUnderstandTheAccessibilityImplications` are a little crazy, much like `classNameAndIHaveSpokenToDST`. I feel like it should just be part of the guidelines as opposed to the prop names when we have these outlier situations; although of course only when we really have zero way of getting a good middle ground (which I think isn't the case here).
I liked what they did here: https://redbubble.design/components/button/examples#buttons-as-links

Reply:
I disagree with the comparison with `classNameAndIHaveSpokenToDST` (though they are similar in both being stupidly long prop names). With className it felt unrealistic and out of touch to expect people to talk to us every time they wanted to override something, whereas this one is more of a warning for people when they are going to go against the guidelines. We've seen a lot of examples of developers using our form components without linking a label (and that case should be more straight forward than this), so I don't think it's unreasonable to say that people are struggling to understand when they are "breaking accessibility" and could use some warning on this.

I also think it's unrealistic to think that developers will be across all of our guidelines or read them all the time when building, even after we fix the huge mess that our documentation is currently. I can give you a super recent example of something that's documented well, and there's an interface that was just built that goes against it for no apparent reason.

If we are going for the appending solution anyway this doesn't matter, but I don't think we should be seeing that prop type as an anti-pattern.